plugins/inputs/nsq/nsq.go

```diff
@@ -65,6 +88,15 @@ func (n *NSQ) Description() string {
 }
 
 func (n *NSQ) Gather(acc telegraf.Accumulator) error {
+	var err error
+
+	n.tlsConfigOnce.Do(func() {
+		n.tlsConfig, err = n.buildTLSConfig()
+	})
+	if err != nil {
+		return fmt.Errorf("fail to build tls config: %v", err)
+	}
```

Comment:
I think this is going to act a little strange if the tls.Config isn't created succesfully, because the second time Gather is called it will not return an error here.  You may want to look at the nginx input for a good example.